crates/red_knot_python_semantic/src/types.rs

```diff
@@ -654,6 +656,11 @@ impl<'db> Type<'db> {
                     .is_subtype_of(db, target)
             }
 
+            (
+                Type::Callable(CallableType::General(self_callable)),
+                Type::Callable(CallableType::General(other_callable)),
+            ) => self_callable.is_subtype_of(db, other_callable),
+
             (Type::Callable(CallableType::General(_)), _) => {
                 // TODO: Implement subtyping for general callable types
```

Comment:
Seems like we can get rid of this TODO now? Or at least replace it with something more specific?
```suggestion
                // TODO: Implement subtyping between general callable types and other types (function literals, class literals, `type[]`...)
```